ReactAndroid/src/main/java/com/facebook/react/processing/ReactNativeModuleProcessor.java

```diff
@@ -0,0 +1,430 @@
+
+      return builder.toString();
+    }
+
+    private char paramTypeToChar(VariableElement parameter) {
+      TypeName indexType = TypeName.get(parameter.asType());
+      char tryCommon = commonTypeToChar(indexType);
+      if (tryCommon != '\0') {
+        return tryCommon;
+      }
+      if (ClassName.get(parameter.asType()).equals(EXECUTORTOKEN)) {
+        return 'T';
+      } else if (indexType.equals(TypeName.get(Callback.class))) {
+        return 'X';
+      } else if (indexType.equals(TypeName.get(Promise.class))) {
+        return 'P';
+      } else if (indexType.equals(TypeName.get(ReadableMap.class))) {
+        return 'M';
+      } else if (indexType.equals(TypeName.get(ReadableArray.class))) {
+        return 'A';
+      } else {
+        throw new RuntimeException(
+          "Got unknown param class: " + indexType.toString());
+      }
+    }
+
+    private char commonTypeToChar(TypeName indexType) {
+      if (indexType.equals(TypeName.get(boolean.class))) {
+        return 'z';
+      } else if (indexType.equals(TypeName.get(Boolean.class))) {
+        return 'Z';
+      } else if (indexType.equals(TypeName.get(int.class))) {
+        return 'i';
+      } else if (indexType.equals(TypeName.get(Integer.class))) {
+        return 'I';
+      } else if (indexType.equals(TypeName.get(double.class))) {
+        return 'd';
+      } else if (indexType.equals(TypeName.get(Double.class))) {
+        return 'D';
+      } else if (indexType.equals(TypeName.get(float.class))) {
+        return 'f';
+      } else if (indexType.equals(TypeName.get(Float.class))) {
+        return 'F';
+      } else if (indexType.equals(TypeName.get(String.class))) {
+        return 'S';
+      } else {
+        return '\0';
+      }
+    }
+
+    public CodeBlock createInvokeMethod() {
+      CodeBlock.Builder builder = CodeBlock.builder();
+      builder.indent().add("if($L != $L.size()) {\n", mParametersNum, "parameters");
+      builder.indent().add("throw new $T($L.getClass().getName()+\".\"+$S+\" got \"+$L.toString()+\"arguments, except \"+$L);\n", NATIVE_ARGUMENTS_PARSE_EXCEPTION, FIELD_NAME_BASE_JAVA_MODULE, mName, "parameters", mParametersNum);
+      builder.unindent().add("}\n");
+      builder.add("$L.$L(", FIELD_NAME_BASE_JAVA_MODULE, mName);
+      int offset = 0;
+      for (int i = 0, len = mParameters.size(); i < len; i++) {
+        TypeName indexType = TypeName.get(mParameters.get(i).asType());
+        if ((ClassName.get(mParameters.get(i).asType()).equals(EXECUTORTOKEN))) {
+          builder.add("$L", "executorToken");
+          builder.add(", ");
+          continue;
+        } else if (indexType.equals(TypeName.get(boolean.class)) || indexType.equals(TypeName.get(Boolean.class))) {
+          builder.add("$L.getBoolean($L)", "parameters", offset);
+        } else if (indexType.equals(TypeName.get(int.class)) || indexType.equals(TypeName.get(Integer.class))) {
+          builder.add("$L.getInt($L)", "parameters", offset);
```

Comment:
er..., it is strange to use getDouble and convert to int or float. I have read the original code of ARGUMENT_EXTRACTOR_INTEGER and ARGUMENT_EXTRACTOR_FLOAT. I will fix this and refresh PR